### PR TITLE
Cif 751 add to cart button

### PR DESCRIPTION
**Pull request by AKumarSingh**
CIF 751 add to cart button integration

## Description

Add to cart button should add the items to cart as described in CIF-751 
https://jira.corp.adobe.com/browse/CIF-751 

## How Has This Been Tested?

Go to any product page:
Select variant ,  color if applicable and click add to cart button. Cart opens and Item gets added in cart. 

## Types of changes

Minor change to integrate product page button and cart. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js [hunk 1]
```diff
@@ -69,6 +69,11 @@ let addToCartCtx = (function(document) {
             const quantity = document.querySelector(AddToCart.selectors.quantity).value;
 
             if (this._state.sku) {
+                if (window.CIF && window.CIF.MiniCart) {
+                    window.CIF.MiniCart.addItem({ sku: this._state.sku, qty: quantity });
+                } else {
+                    console.log(`Mini cart not available !Ensure that you  have used the correct template for product`);
+                }
                 console.log(
```
Comment (dplaton) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js]: This log message is not needed here.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js [hunk 2]
```diff
@@ -69,6 +69,11 @@ let addToCartCtx = (function(document) {
             const quantity = document.querySelector(AddToCart.selectors.quantity).value;
 
             if (this._state.sku) {
+                if (window.CIF && window.CIF.MiniCart) {
+                    window.CIF.MiniCart.addItem({ sku: this._state.sku, qty: quantity });
+                } else {
+                    console.log(`Mini cart not available !Ensure that you  have used the correct template for product`);
```
Comment (herzog31) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js]: > `MiniCart component not available! Ensure that you have used the correct template.`

File: ui.apps/src/test/components/commerce/product/addToCartTest.js [hunk 3]
```diff
@@ -112,7 +112,11 @@ describe('Product', () => {
 
         it('adds a product to the cart on click', () => {
```
Comment (herzog31) [inline ui.apps/src/test/components/commerce/product/addToCartTest.js]: This test should check that if you click on the button, that `window.CIF.MiniCart.addItem` was called. You can remove the spy on the `_onAddToCart` method.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js (hunk 2 above)
Comment (AKumarSingh) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/addToCart.js]: this is done

File: ui.apps/src/test/components/commerce/product/addToCartTest.js (hunk 3 above)
Comment (AKumarSingh) [reply ui.apps/src/test/components/commerce/product/addToCartTest.js]: done

**Pull request merged**